actor-tests/src/test/scala/org/apache/pekko/pattern/PatternSpec.scala

```diff
@@ -76,4 +74,22 @@ class PatternSpec extends PekkoSpec {
       intercept[IllegalStateException] { Await.result(r, remainingOrDefault) }.getMessage should ===("Mexico")
     }
   }
+
+  "pattern.timeout" must {
+    "be completed successfully eventually" in {
+      val f = pekko.pattern.timeout(1.second, using = system.scheduler)(Future.successful(5))
+      Await.result(f, remainingOrDefault) should ===(5)
+    }
+
+    "be completed abnormally eventually" in {
+      val f = pekko.pattern.timeout(1.second, using = system.scheduler)(Future.failed(new IllegalStateException("ABC")))
+      intercept[IllegalStateException] { Await.result(f, remainingOrDefault) }.getMessage should ===("ABC")
+    }
+
+    "be completed with a TimeoutException if not completed within the specified time" in {
+      val f = pekko.pattern.timeout(1.second, using = system.scheduler)(Future.never)
```

Comment:
Can you use a shorter timeout? I don't see why the test needs to always take 1 second (or more). I think you should also check that the elapsed time for the test is greater than or equal to the timeout - maybe with a small delta (allow a few milliseconds leeway).